### DISCUSSION TITLE
ways we could refactor the bark editor

### DISCUSSION THREAD

**Discussion by dumorando** [cytrinc/bark-coding#200] [Admin General]
1. move to vite :trollface: 
2. move each block to its own file https://rentry.co/zzq37m7k
3. move toolbox to a fetch request https://rentry.co/mndbspci
4. move other javascript to its own file since it takes up so much space in the html

idk chat what do u think?

**Comment by raynepaws** [comment]
good idea getting the js out of the way, it's very annoying. however, why do we need these other things?

**Reply by dumorando** [reply]
to make the code easier to read   
so like the blocks you dont have to scroll through one massive js file   
and the toolbox isnt in the html, its in toolbox.xml or smth to keep blockly stuff out of the html
